### PR TITLE
Handle paths with spaces on windows for ctags

### PR DESCRIPTION
**Pull request by johatfie**
ctags now runs correctly in the background and handles paths with spaces.

Sorry for the messy commit history.  I'm still pretty new to git and didn't think to do all this in a branch other than master.

### CODE REVIEW

File: autoload/gutentags/ctags.vim [hunk 1]
```diff
@@ -44,7 +44,7 @@ function! gutentags#ctags#generate(proj_dir, tags_file, write_mode) abort
 
     try
         " Build the command line.
-        let l:cmd = gutentags#get_execute_cmd() . s:runner_exe
+        let l:cmd = gutentags#get_execute_cmd() . '"' . s:runner_exe . '"'
```
Comment (justinmk) [inline autoload/gutentags/ctags.vim]: Don't do this. `shellescape()` exists for a reason, you'll run into more edge cases which were already solved by `shellescape()`.

**Comment by johatfie** [comment]
Thanks.  I'm pretty new to vim script as well.

**Comment by ludovicchabant** [comment]
Thanks! It's going to take me some time to test this so hang in there :)

**Comment by johatfie** [comment]
So, are you going to test this?  It's been two months.  :)

**Comment by ludovicchabant** [comment]
I've been quite busy.
What's this "WIP" commit?

**Comment by johatfie** [comment]
WIP = work in progress.  I know it's a messy commit history.  I'm still very new to git.

**Comment by ludovicchabant** [comment]
Can't help you there, I also often mess up with Git (which is why I use Mercurial and mirror stuff on Github via hg-git). But yeah the commit history is too messy to merge right now, sorry. I don't know what the best way to clean it is, short of re-doing it :(

**Comment by johatfie** [comment]
Oh, I just figured you'd do a merge --squash.  But, I can redo it.  It should have been in a separate branch anyway.

**Comment by ludovicchabant** [comment]
Closing it since it's been a while.

**Pull request closed without merging**